### PR TITLE
Go Intro for Master branch

### PR DESCRIPTION
**Pull request by CitMC**
Documentation content update: Adding Go Intro chapter

### Change scope

The changes in this Pull Request include (please tick all that apply):

- [ ] Small language/grammar fixes
- [ ] Small content fixes 
- [x] Addition of new content
- [ ] Sample code/command updates
- [ ] Platform fixes
- [ ] Other

### R1: Internal review (B9lab)

- Technical review
  - [ ] requested
  - [ ] completed
- Language review
  - [x] requested
  - [x] completed

### R2: External review

- Technical review
  - [x] requested
  - [x] completed
- Language review
  - [x] requested
  - [x] completed

### R3: Internal QA review (B9lab)

- Technical review
  - [ ] requested
  - [ ] completed
- Language review
  - [ ] requested
  - [ ] completed

### R4: Final external QA review

- Technical review
  - [ ] requested
  - [ ] completed
- Language review
  - [ ] requested
  - [ ] completed

### RC: Release Candidate

- [ ] Ready to be merged

### CODE REVIEW

File: academy/golang-intro/1_install.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+</HighlightBox>
+
+Now, let's compile this program. Create the file `hello.go` in the `$GOPATH/src/hello` directory. In a terminal:
+
+```
+$ cd $GOPATH/src/hello
+$ go build
+```
+
+If all goes well, `go` builds the executable. Go only prints errors. So, you should see no output and another command prompt unless something goes wrong.
```
Comment (AndrewB9lab) [inline academy/golang-intro/1_install.md:67]: There is an ambiguity in "you should see **_no output and another command prompt_** unless something goes wrong."

Options: 

1. **One** event occurs: you see no output; **then a new command prompt appears**
2. **Zero** events occur: you see no output; and you see no new command prompt

I suspect the correct intention is option 1. Confirm please!

File: academy/golang-intro/1_install.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+func main() {
+    fmt.Printf("Hello, World!\n")
+}
+```
+
+[Test online](https://go.dev/play/p/1u5bSZlh80h)
+
+Every Go program starts with the package declaration. Each package consists of one or more Go source files in a single directory. If the name of the package is `main`, Go creates an executable file.
+
+After that, import the package `fmt`. Note that the package name is the last element of the package's file path. For an example, if you import the package "lib/math", then you use it as "math".
```
Comment (AndrewB9lab) [inline academy/golang-intro/1_install.md:46]: It is not clear what "After that" refers to. 

Suggestion: "After running the **_Hello, World!_** program, ..."

Confirm please!

File: academy/golang-intro/1_install.md (hunk 1 above)
Comment (murataniloener) [reply academy/golang-intro/1_install.md:67]: 2. so an "or" instead of "and" I guess :)

Comment (AndrewB9lab) [reply academy/golang-intro/1_install.md:67]: I'm going with the very exciting "nor" for this! ;-) Thanks!

File: academy/golang-intro/6_packages.md [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+
+If you use an invalid verb, you will get a string beginning with `"%!"` and a description.
+
+To write to `stdout`:
+
+1. `func Print(a ...interface{}) (n int, err error)` formats with default formats.
+2. `func Println(a ...interface{}) (n int, err error)` formats with default formats and appends a newline.
+3. `func Printf(format string, a ...interface{}) (n int, err error)` formats with given format.
+
+These 2 functions return the number of bytes written and any applicable error.
```
[the remark this answers is not in the document]
Comment (murataniloener) [reply academy/golang-intro/6_packages.md:34]: Good catch! Yes there are 3 functions, you can correct it directly without and extension.

File: academy/golang-intro/1_install.md (hunk 1 above)
Comment (AndrewB9lab) [reply academy/golang-intro/1_install.md:67]: My edit will show "you should not see any output nor another command prompt" - thanks!

File: academy/golang-intro/7_concurrency.md [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+
+<HighlightBox type="tip">
+
+[Test it online](https://go.dev/play/p/MYdZRhyG36y)
+
+</HighlightBox>
+
+Run this programm. What happened there?
+
+We don't need to use `time.Sleep` anymore because sends and receives block until the other side is ready. To avoid blocking, one can create *buffered* channels.
```
Comment (AndrewB9lab) [inline academy/golang-intro/7_concurrency.md:109]: In **109**, I am reading the word "block" as if it is being used in the sense of "prevent something happening" (same with "blocking"). 

However, in **115**, we have the following: _"When a *buffered* channel is full, **sends block**. When and if it is empty the channel **receives blocks**."_ This use of "blocks" reads more like the idea of _blockchain_ blocks, as in "a collection of data which is sent or received".

Are we using two different senses of the word "block" in close proximity here? If not, which is the sense both lines should be understood as?

Comment (AndrewB9lab) [reply academy/golang-intro/7_concurrency.md:109]: So, for 115:

When a *buffered* channel is full, sends to it are blocked. When a buffered channel is empty, receives from it are blocked.

Does that read appropriately?

Comment (murataniloener) [reply academy/golang-intro/7_concurrency.md:109]: Perfect!

File: academy/golang-intro/1_install.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+Start with the basics. In this section, you will install Go and learn some general things about it.
+
+
+## Install
+
+Installing Go depends on your machine. See the [Getting Started Page](https://golang.org/doc/install) and follow the instructions to install Go.
+
+<HighlightBox type="tip">
+
+[Gimme](https://github.com/travis-ci/gimme) is a useful script for Go.
```
Comment (julienrbrt) [inline academy/golang-intro/1_install.md:21]: This does not look maintained anymore. I think people should stick with the official docs you've mentioned just above.

Comment (julienrbrt) [inline academy/golang-intro/1_install.md:21]: This does not look maintained anymore. I think people should stick with the official docs you've mentioned just above.

File: academy/golang-intro/1_install.md [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+
+</HighlightBox>
+
+## Workflow
+
+Go has a different workflow from most other programming tools. You work with a single workspace and have all Go code contained in this workspace. The `GOPATH` environment variable specifies the location of your workspace. You can print it by typing `$ go env GOPATH` in your console. For more information about `GOPATH` type `$ go help gopath`.
```
Comment (julienrbrt) [inline academy/golang-intro/1_install.md:27]: This make it seem like all your project needs to be in `GOPATH/src/` which is not the case.
You can have your code anywhere, but go will the dependencies in your `GOPATH`.

Comment (julienrbrt) [inline academy/golang-intro/1_install.md:27]: This make it seem like all your project needs to be in `GOPATH/src/` which is not the case.
You can have your code anywhere, but go will the dependencies in your `GOPATH`.

File: academy/golang-intro/1_install.md [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+
+<HighlightBox type="tip">
+
+[Test it online](https://go.dev/play/p/1u5bSZlh80h).
+
+</HighlightBox>
+
+<HighlightBox type="info">
+
+Function names determine visibility outside of packages. `Printf`'s first character is upper case and this means it's visible outside of `fmt`. Package names are always written in lower case, like `fmt` (short for "format").
```
Comment (julienrbrt) [inline academy/golang-intro/1_install.md:52]:
```suggestion
Function names determine visibility outside of packages. `Printf`'s first character is upper case and this means it's visible outside of `fmt`– a.k.a public. Package names are always written in lower case, like `fmt` (short for "format").
```
```suggestion
Function names determine visibility outside of packages. `Printf`'s first character is upper case and this means it's visible outside of `fmt`. Package names are always written in lower case, like `fmt` (short for "format").
```

Comment (julienrbrt) [inline academy/golang-intro/1_install.md:52]:
```suggestion
Function names determine visibility outside of packages. `Printf`'s first character is upper case and this means it's visible outside of `fmt`– a.k.a public. Package names are always written in lower case, like `fmt` (short for "format").
```
```suggestion
Function names determine visibility outside of packages. `Printf`'s first character is upper case and this means it's visible outside of `fmt`. Package names are always written in lower case, like `fmt` (short for "format").
```

File: academy/golang-intro/1_install.md [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+
+After package declaration, import the package `fmt`. Note that the package name is the last element of the package's file path. For example, if you import the package "lib/math", then you use it as "math".
+
+`fmt` implements input and output, and is part of Go's standard library. Later you are going to describe some of its functions in more detail.
+
+Execution starts in the function called `main()`.
+
+This function simply calls the Go I/O function `Printf()` from the package `fmt`.
+
+Now compile this program. Create the file `hello.go` in the `$GOPATH/src/hello` directory. In a terminal:
```
Comment (julienrbrt) [inline academy/golang-intro/1_install.md:66]: This is not necessary anymore to have the file here. We should just tell them to create a folder somewhere of their choice.

Comment (julienrbrt) [inline academy/golang-intro/1_install.md:66]: This is not necessary anymore to have the file here.

File: academy/golang-intro/1_install.md [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+
+```
+$ cd $GOPATH/src/hello
```
Comment (julienrbrt) [inline academy/golang-intro/1_install.md:69]:
```suggestion
$ go mod init hello
```

Comment (julienrbrt) [inline academy/golang-intro/1_install.md:69]:
```suggestion
$ go mod init hello
```

**Review by julienrbrt — APPROVED** [review]
lgtm, just a few nits above.

File: academy/golang-intro/1_install.md (hunk 5 above)
Comment (CitMC) [reply academy/golang-intro/1_install.md:21]: Tackled in the last commit.

File: academy/golang-intro/1_install.md (hunk 6 above)
Comment (CitMC) [reply academy/golang-intro/1_install.md:27]: Tackled in the last commit with minor edit.

File: academy/golang-intro/1_install.md (hunk 8 above)
Comment (CitMC) [reply academy/golang-intro/1_install.md:66]: Tackled in the last commit with minor edit.

**Pull request merged**